common/system/tasks/radio_manager.cpp

```diff
@@ -14,99 +14,124 @@
 #include "data_log.h"
 #include "data_structures.h"
 #include "hardware_manager.h"
+#include "packet_encoder.h"
 
 #define RADIO_INTERVAL_MS 200
 #define RADIO_SEND_MS 100
 
-static DataRecieveState_s dataRx[NUM_RADIO];
-static DataTransmitState_s lastSent[NUM_RADIO];
-
-static RadioPacket_s transmitPacket[NUM_RADIO];
-
 static const char *call = "KM6GNL";
 
 // https://stackoverflow.com/q/9695329
 #define ROUND_2_INT(f) ((int)((f) >= 0.0 ? (f + 0.5) : (f - 0.5)))
 
-void radioManager_init() {
+// this is gross and smelly, but for rn, its valid that one board has only one
+// kind of radio on it. In the future, refactor based on transmit mode to
+// support lora as well
+#if HAS_DEV(RADIO_TI_433) || HAS_DEV(RADIO_TI_915)
+FSKPacketRadioEncoder packetEncoder;
+#else
+PassthroughRadioEncoder packetEncoder;
+#endif
+
+/**
+ * calculate the CRC of a packet, up to but excluding the CRC field at the end
+ */
+static uint16_t calculateRadioPacketCRC(RadioDecodedPacket_s &packet) {
+  // Init value for CRC calculation, by convention
+  uint16_t checksum = 0xFFFF;
+
+  // iterate over whole packet, up until CRC
+  uint8_t *pktAsBytes = reinterpret_cast<uint8_t *>(&packet);
```

Comment:
Is the linter somehow not running on this file? `PointerAlignment: Left` should be complaining here (and some other places in this file).

Reply:
Or wait, we aren't consistent with this throughout files already outside of this PR? Not sure what's up with that

Reply:
skull yeah that should be in its own PR